api.md

```diff
@@ -287,6 +287,12 @@ in the table.
 to how you handle constraints. A hypertable can contain foreign keys to normal SQL table
 columns, but the reverse is not allowed. UNIQUE and PRIMARY constraints must include the
 partitioning key.
+
+>:WARNING: Running `create_hypertable` with data migration on a table with foreign key contraints to other tables 
+might result in a deadlock if parallel transactions simulteniously try to insert data into the table and 
+the referenced tables. This can result in aborting `create_hypertable`. The deadlock can be prevented by 
+manually obtaining `SHARE ROW EXCLUSIVE` lock on the referenced tables  before calling `create_hypertable` 
+in the same transaction.
```

Comment:
Hmm... quite hard to read. Too many indirect references and digressions. Here is a slightly improved version, but I think it need some more work.

> >:WARNING: If you run `create_hypertable` on a table with foreign key constraints 
> you might have a deadlock. This can happen if other transactions try to insert data into the foreign table at the same time as the `create_hypertable` is running. As a result, the `create_hypertable` will be aborted.
> >
> > You can prevent the deadlock by executing `SHARE ROW EXCLUSIVE` in the same transaction as `create_hypertable`.

Reply:
It is difficult to disagree that my formulation is too complicated :)
I made some improvements.